### ISSUE TITLE
Wrong results in a very basic convolution on CPU (results are correct on GPU)!

### ISSUE THREAD

**Issue by xxtemp** [pytorch/pytorch#18948]
## 🐛 Bug

Running a 2d convolution (no padding, stride=1) on a 3x1 input, sliced from a 3x3 tensor, and a 3x1 weight, sliced from a 3x3 tensor. When trying to run it on CPU the convolution result is wrong, but when moving to GPU (by adding .cuda() after the tensors), the result is correct!

## To Reproduce

Run the code below to reproduce the bug. It's pretty self explanatory. 

```
import torch
import torch.nn.functional as F

x = torch.Tensor([[[[0., 2., 2.],
                    [0., 0., 1.],
                    [2., 0., 0.]]]])
w = torch.Tensor([[[[1., 1., 0.],
                    [1., 0., 3.],
                    [0., 3., 1.]]]])

x1 = x[:,:,:,2:] # Using only the last column from x, containing [2,1,0]
w1 = w[:,:,:,2:] # Using only the last column from w, containing [0,3,1]
#The result should be ((2*0) + (1*3) + (0*1)) = 3

print(F.conv2d(x1, w1, None, padding=0, stride=1)) # tensor([[[[1.]]]]) WRONG!!!!
print(F.conv2d(x1.cuda(), w1.cuda(), None, padding=0, stride=1)) # tensor([[[[3.]]]], device='cuda:0') as expected
```

## Expected behavior

As seen in the code above, the CPU gives the result of 1, when the actual result should be 3.

## Environment

 - PyTorch Version (e.g., 1.0): 1.0
 - OS (e.g., Linux): Ubuntu 16.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): 
 - Python version: 3.5.2
 - CUDA/cuDNN version: 9.0
 - GPU models and configuration: GeForce GTX 1060
 - Any other relevant information:

**Comment by S-Ellis** [comment]
The code runs correctly for me (Ubuntu 16.04, pytorch 1.0.1, conda install, python 3.6)

**Comment by fmassa** [comment]
This is a duplicate of https://github.com/pytorch/pytorch/issues/16018 and has been fixed in 1.0.1
